include/ChaiVM/interpreter/code-manager/code-manager.hpp

```diff
@@ -0,0 +1,113 @@
+#pragma once
+
+#include <cassert>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <istream>
+#include <vector>
+
+#include "ChaiVM/interpreter/code-manager/func-struct.hpp"
+#include "ChaiVM/memory/allocator.hpp"
+#include "ChaiVM/memory/linear-allocator.hpp"
+#include "ChaiVM/memory/linear-buffer.hpp"
+
+namespace chai::interpreter {
+
+/**
+ * Class to manage bytecode. Similar to ClassLoader.
+ */
+class CodeManager final {
+public:
+    enum ConstantTag : uint8_t {
+        CNST_I64 = 1,
+        CNST_F64,
+        CNST_FUNC_NAME_AND_TYPE,
+        CNST_RAW_STR, // Analogue of CONSTANT_Utf8 in jvm
+    };
+
+    /**
+     * Parses and loads the full file.
+     * @param path Path to the .chai file.
+     */
+    void load(const std::filesystem::path &path);
+
+    /**
+     * Loads stream of Constant pool.
+     * @param istream
+     */
+    void loadPool(std::istream &istream);
+
+    void loadFunction(std::istream &istream) {
```

Comment:
Why did you implement this function in `code-manager.hpp`? Could you please move this implementation to `code-manager.cpp`.